src/rate-limit.coffee

```diff
@@ -53,27 +55,39 @@ module.exports = (robot) ->
       else
         roomOrUser = context.response.message.user.name
 
-      # Construct a key to rate limit on. If the response was from a room
-      # then append the room name to the key. Otherwise, append the user name.
-      listenerAndRoom = listenerID + "_" + roomOrUser
+      # Get the key used for storing data in lastExecutedTime and
+      # lastNotifiedTime
+      if context.listener.options.rateLimits &&
+         context.listener.options.rateLimits.keyFunction
+        # Get key by calling a user-defined function
+        robot.logger.debug "Getting key from keyFunction"
```

Comment:
I think these are a little too verbose (will result in an additional, identical log line for every single inbound message that matches the listener regex).